Dynavity/Dynavity/model/canvas/Canvas.swift

```diff
@@ -1,75 +1,46 @@
+import Combine
 import Foundation
 import CoreGraphics
 
-struct Canvas {
-    var canvasElements: [CanvasElementProtocol] = []
-    var umlConnectors: [UmlConnector] = []
+class Canvas: ObservableObject {
+    @Published private(set) var canvasElements: [CanvasElementProtocol] = []
+    @Published private(set) var umlConnectors: [UmlConnector] = []
     var name: String = "common"
+    private var cancellables: [AnyCancellable] = []
 
-    mutating func addElement(_ element: CanvasElementProtocol) {
+    func addElement(_ element: CanvasElementProtocol) {
         canvasElements.append(element)
-    }
-
-    mutating func removeElement(_ element: CanvasElementProtocol) {
-        guard let index = canvasElements.firstIndex(where: { $0.id == element.id }) else {
-            return
-        }
-
-        canvasElements.remove(at: index)
-    }
-
-    mutating func replaceElement(_ element: CanvasElementProtocol) {
-        guard let index = canvasElements.firstIndex(where: { $0.id == element.id }) else {
-            return
+        let cancellable = element.objectWillChange.sink { [weak self] _ in
+            self?.objectWillChange.send()
         }
-
-        canvasElements[index] = element
-    }
-
-    func getElementBy(id: UUID?) -> CanvasElementProtocol? {
-        canvasElements.first(where: { $0.id == id })
+        cancellables.append(cancellable)
     }
 
-    mutating func moveCanvasElement(id: UUID?, by translation: CGSize) {
-        guard let index = canvasElements.firstIndex(where: { $0.id == id }) else {
+    func removeElement(_ element: CanvasElementProtocol) {
+        guard let index = canvasElements.firstIndex(where: { $0 === element }) else {
             return
         }
 
-        canvasElements[index].move(by: translation)
-    }
-
-    mutating func resizeCanvasElement(id: UUID?, by translation: CGSize) {
-        guard let index = canvasElements.firstIndex(where: { $0.id == id }) else {
-            return
-        }
-
-        canvasElements[index].resize(by: translation)
-    }
-
-    mutating func rotateCanvasElement(id: UUID?, to rotation: Double) {
-        guard let index = canvasElements.firstIndex(where: { $0.id == id }) else {
-            return
-        }
-
-        canvasElements[index].rotate(to: rotation)
+        canvasElements.remove(at: index)
+        cancellables.remove(at: index)
```

Comment:
I think this may be quite prone to bugs. Maybe we can rename cancellables to canvasElementsCancellables, so we don't add other stuff to it